src/view/view.rs

```diff
@@ -41,14 +59,14 @@ use crate::{
 /// and also a type for actions which are passed up the tree in event
 /// propagation. During event handling, mutable access to the app state is
 /// given to view nodes, which in turn can make expose it to callbacks.
-// We depend on ViewMarker to disallow any View implementations which dont implement ViewMarker.
+// We depend on ViewMarker to disallow any View implementations which don't implement ViewMarker.
 // Doing that would lead to pretty strange bugs.
-pub trait View<T, A = ()>: ViewMarker + Send {
+pub trait GenericView<T, W: ?Sized, A = ()>: ViewMarker + Send {
     /// Associated state for the view.
     type State: Send;
 
     /// The associated widget for the view.
-    type Element: Widget;
+    type Element: TraitBound<W>;
```

Comment:
I'd like to see some better documentation on the generic parameters in this trait (this PR seems to introduce a new one without explaining it, and `A` is never mentioned by name)